applications/CoSimulationApplication/python_scripts/data_transfer_operators/sum_distributed_to_distributed.py

```diff
@@ -0,0 +1,43 @@
+# Importing the base class
+from KratosMultiphysics.CoSimulationApplication.base_classes.co_simulation_data_transfer_operator import CoSimulationDataTransferOperator
+
+# Other imports
+import numpy as np
+
+def Create(settings):
+    return SumDistributedToDistributed(settings)
+
+class SumDistributedToDistributed(CoSimulationDataTransferOperator):
+    """DataTransferOperator to sum values on one interface and put it to another interface, if desired equally distributed.    
+    """
+    def _ExecuteTransferData(self, from_solver_data, to_solver_data, transfer_options):        
+        if not to_solver_data.IsDefinedOnThisRank():
+            return
```

Comment:
this is wrong

Imagine `from_solver` is using this rank then you deadlock when doing `SumAll`

you have to do some more checks, see the other classes